### DISCUSSION TITLE
Asking community for help on swapping 10% of both staking and treasury rewards to BTC on Hydration, so they both get dual rewards

### DISCUSSION THREAD

**Discussion by SasoLithops** [ktondao/KtonDAO#36] [Q&A]
Based on:

This poll: https://github.com/orgs/darwinia-network/discussions/1577#discussion-7095035
This reply: https://github.com/orgs/darwinia-network/discussions/1577#discussioncomment-11149843
This KTON DAO treasury account: https://explorer.darwinia.network/address/0x08837De0Ae21C270383D9F2de4DB03c7b1314632?tab=internal_txns
**_This ongoing debate_**: https://github.com/orgs/darwinia-network/discussions/1627
And this initial discussion: https://github.com/orgs/darwinia-network/discussions/1566#discussion-7052819

___________________________________________________________________________________________________________________________________________________

**IMPORTANT: At this time KTON stakers get 19000 RING while KTON DAO treasury gets 4650 RING every 6 hours.**

**_ALSO: We are working on building an Index token which will highly benefit KTON stakers and governance and Github participants(Alpha). Any opinion on how we could implement it in staking rewards or its fees to KTON stakers, LP providers is welcome(check link above about ongoing debate). Most important KTON will be part of index token._**

______________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________
______________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________

## This thread is about swapping 10% of both staking and treasury rewards to BTC on Hydration, so they both get dual rewards

Swapping **10% of both the 2,325 RING** sent to the **KTON DAO treasury** and the **19,000 RING** sent to **stakers** into **BTC** on Hydration is a potential strategy to **diversify the treasury**, **stabilize the value of the assets**, and **reward stakers in dual tokens** (RING and BTC). However, it requires careful consideration of several factors to ensure that the move aligns with the **long-term goals of the KTON DAO** and does not negatively impact liquidity or ecosystem health.

Here’s a breakdown of key considerations and an analysis of whether this approach is advisable:

### **1. Objectives of Swapping RING to BTC**
The primary benefits of converting **10% of RING** into **BTC** could include:
- **Treasury Diversification**: Adding BTC to the treasury helps mitigate the potential risks of relying solely on RING as a treasury asset. BTC is a more stable store of value, and holding BTC can protect the DAO against inflationary pressures and volatility in RING’s price.
- **Stabilizing the Treasury**: BTC tends to be less volatile than many altcoins (including RING). By holding BTC, the DAO can create more **stability in its treasury** and ensure it has a valuable, widely accepted asset that can be liquidated or used in DeFi strategies.
- **Dual Token Rewards for Stakers**: Providing rewards in **dual tokens** (RING and BTC) can make staking more attractive to participants, offering them both the potential upside of RING and the stability of BTC. This can increase **staker engagement** and **long-term loyalty**.

However, there are also risks and trade-offs that must be carefully evaluated.

---

#### **A. Impact on DAO Treasury and Liquidity**
1. **Impact on Treasury Liquidity**:
   - **232 RING** from the **DAO treasury** (10% of 2,325 RING) and **1,900 RING** from **staking rewards** (10% of 19,000 RING) will be converted to BTC.
   - If the DAO uses **RING** for specific operational needs (e.g., further liquidity provisioning, governance incentives, etc.), reducing its **RING holdings** could limit its ability to fulfill these needs.
   - Consider whether holding **BTC** in the treasury could create more **flexibility** for future **investment opportunities** or **partnerships**, as BTC is more universally accepted and could be utilized in a wide range of DeFi protocols or exchanges.

2. **Liquidity Risks**:
   - **Liquidity for Stakers**: The DAO must ensure that it still has enough **RING liquidity** available for **staking rewards**, governance incentives, and for maintaining sufficient liquidity for **RING trades** on decentralized exchanges (DEXs). If too much RING is swapped for BTC, there could be a **shortfall** in the amount of RING available for other ecosystem functions.
   - **Liquidity for RING/BTC Pair**: The DAO will need to ensure that it has adequate liquidity in the **KTON/RING and RING/BTC pairs** on exchanges to facilitate trading without causing significant slippage when users want to exit or swap RING for BTC or vice versa.

#### **B. Market Conditions for RING and BTC**
1. **RING Market Conditions**:
   - Evaluate whether **RING** is in a **bullish or neutral phase**. If RING is in a **bullish phase**, converting 10% to BTC could mean missing out on potential upside gains. However, if RING is highly volatile or in a **downtrend**, converting some of the rewards to BTC might be a smart move to hedge against further RING depreciation.
   - If **RING is experiencing inflation** due to continuous reward distributions (like the 19,000 RING every 6 hours), converting some of the RING into BTC can reduce the pressure on RING’s price and create **deflationary pressure** in the RING market.

2. **BTC Market Conditions**:
   - BTC is typically **more stable** than RING, but it is still subject to market cycles. The DAO should consider the **current BTC price** and **market trend** (e.g., is BTC in a consolidation phase or a bull market?). Timing the conversion could have an impact on how much BTC the DAO ultimately acquires. 
   - It might be beneficial to perform the swap **gradually** over time (e.g., monthly or quarterly) to avoid buying BTC at unfavorable market prices and to reduce the risk of **timing issues**.

#### **C. Costs of Converting RING to BTC on Hydration**
1. **Swap Fees and Slippage**:
   - **Hydration fees**: Ensure that the **swap fees** on Hydration are reasonable for converting RING to BTC. High fees or slippage could erode the value of the swap.
   - **Liquidity on Hydration**: Check that there is enough liquidity in the RING/BTC pool on Hydration to facilitate the swaps. Low liquidity could result in **slippage**, meaning the DAO may not get the full value it expects for its RING tokens when swapping for BTC.

2. **Transaction Speed and Costs**:
   - Consider the **gas fees** or transaction costs involved in executing these swaps. If RING and BTC are on different chains (e.g., Ethereum for RING and Bitcoin for BTC), cross-chain swaps could incur additional costs, which should be factored into the overall calculation.

---

#### **A. Proportional Allocation**
The proposal to swap **10%** of the RING rewards to BTC is a **moderate approach** that helps diversify the treasury and staker rewards without over-exposing the DAO to either RING or BTC.

- **10% of Treasury RING Rewards**: **232 RING** from the **2,325 RING** allocated to the DAO treasury can be swapped into BTC.
- **10% of Staking Rewards**: **1,900 RING** from the **19,000 RING** allocated to stakers can be swapped into BTC.

This keeps the conversion proportionate and consistent across both the DAO’s treasury and the staking rewards pool, allowing the DAO to **stabilize the treasury** and provide **dual token rewards** to stakers without risking liquidity shortfalls.

#### **B. Consider Staggering the Swaps**
Rather than doing a single large swap, the DAO could execute **swaps in smaller increments** over time to reduce the impact of **market timing** and to mitigate risks of **slippage** or **unfavorable pricing**.

For example:
- Convert **2.5%** of RING rewards into BTC each week or month, instead of a one-off 10% swap. This way, the DAO can assess the impact of each swap and adjust its strategy if needed.

#### **C. Balancing Dual Token Rewards for Stakers**
Rewarding stakers in **both RING and BTC** can make the staking program more attractive, as it provides **dual incentives**:
- **RING** could be distributed as part of staking rewards for **network growth**, governance, and engagement.
- **BTC** offers **stability**, and **BTC rewards** could be seen as a long-term store of value that complements the more volatile RING rewards.

It’s important to ensure that the conversion of 10% of staking rewards into BTC does not **drain liquidity** needed for **daily staking distributions** or cause **confusion** among stakers. Clearly communicate the dual token reward structure, explaining why the DAO is making the conversion and how it benefits the ecosystem.

---

#### **A. Risk of Reducing RING’s Liquidity**
- **Mitigation**: Keep track of how much RING is being swapped and ensure there is always enough RING available for **staking rewards** and **liquidity provisioning** on decentralized exchanges. A **reduction of too much RING** could reduce liquidity and impact stakers' ability to redeem their rewards.

#### **B. Market Timing Risks**
- **Mitigation**: Use a **staggered conversion schedule** to avoid unfavorable market conditions (e.g., when BTC is at a high price or RING is in a downtrend).

#### **C. Swap Fees and Slippage**
- **Mitigation**: Evaluate the **liquidity and fees** on Hydration, or consider using a more liquid platform if Hydration does not provide competitive rates or sufficient liquidity.

---

### **Conclusion:**
Swapping **10% of both the 2,325 RING rewards to the DAO treasury** and the **19,000 RING staking rewards** to **BTC** on Hydration can be a sound strategy to **diversify** the DAO’s holdings, **stabilize the treasury**, and reward stakers with **dual tokens**. 

### **Key Recommendations**:
1. **Proportional Allocation**: Stick to the proposed **10%** swap to balance liquidity, treasury needs, and reward distribution.
2. **Staggered Execution**: Consider gradually executing the swap over time to reduce market timing risks and slippage.
3. **Monitor Liquidity**: Ensure that liquidity for both RING and BTC remains sufficient to avoid shortfalls for staking rewards and liquidity provision.
4. **Clear Communication**: Ensure stakers are informed of the new dual-token reward structure and the rationale behind the BTC reward.

By implementing this approach carefully, the DAO can achieve both **stability** in its treasury and **increased participation** in the staking program.

______________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________
________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________

**Comment by sonoferin2000** [comment]
I fully support this proposal. Diversifying the treasury with a widely accepted asset like BTC is a smart move. Right now, we don’t have many other tokens in the mix, so adding BTC would be a huge step forward. I can't imagine anyone would disagree with this. As for the stakers, I’m sure they’ll be thrilled to receive BTC alongside their RING rewards—it adds a level of stability and a long-term store of value to the mix. The ecosystem could really benefit from this infusion of an alternative token. I'm definitely in favor of moving forward with this proposal.

**Comment by SasoLithops** [comment]
Agree @sonoferin2000

@hackfisher who can help in contract development and what would be the cost? We could use DCA in Hydration and iBTC is one of options. I can dig deeper into this just need to know what is needed for this to work.

**Reply by hackfisher** [reply]
Hi,

In the long term, I support diversifying assets, but I think the timing and plan are currently too complex.

Please put this on hold for now. Based on my evaluation, the complexity of implementing this is quite high, including integration challenges and potential risks. For example, staking collateral should remain untouched. I would prefer authorizing a Treasury Manager (e.g., via a multisig) to handle this, as it would be more convenient. Additionally, the cross-chain usability of parachains is still limited, and I am deeply concerned about whether the technical team has enough capacity to support such complex requirements from KtonDAO. (Considering that they are service providers for RingDAO, and RingDAO already has many tasks, KtonDAO should likely be a lower priority.)

Let’s keep it simple.

**Reply by SasoLithops** [reply]
This was supposed to be the highlight of new KTON 2.0 staking model. Can we do it manually? And buy it from Hydration using DCA. And distribute once a month to stakers, snapshot at random montly date( to keep it simple like we did with PINK tokens distribution ). BTC could be kept in multisig wallet with Treasury manager and 4 other members with 4 needing to sign to leave the wallet. Or some other sollution. I know we ambassadors use Safe wallet and works well.

**Reply by hackfisher** [reply]
Here’s my refined understanding and summary:

Allocate a larger portion of RING income to a KtonDAO Treasury asset diversification plan, to be executed and managed by a Treasury Manager (e.g., via a Safe multisig). (In my view, this alone is sufficient.)

Alternatively, consider going further by distributing the returns as staking rewards to the KTON staking rewards pool. However, this would require contract upgrades and UI support, which I believe is premature at this stage due to the current funding level being insufficient to cover the associated development costs. （In 1 year income in treasury around 7M RING, which is around 15K USD, how much yield will that be? This approach isn’t cost-effective; the math doesn’t add up.)

**Reply by SasoLithops** [reply]
Yes it all returns to not having decent ammount in treasury. Only way this to work would be if RING price spikes in bull run. That is why this idea was super fast on radar so we don't count and put sell pressure on RING alone and at higher ammounts. But DCA slow.

I agree, I should have asked more upfront on how much this integration will cost and take time... I underastimated costs. Maybe some community member can do it cheaper for the love of project;-)

**Comment by 2xJDubs** [comment]
How hard would it be to support iBTC that is supported on Hydration on Darwinian Network?  This way a DCA plan on Hydration could work, would need to be a multi-sig controlled account for security, and then purchased BTC could be moved to Darwinia for treasury holding.

**Reply by SasoLithops** [reply]
@hackfisher @hujw77 Could you please reply on this. We all interested how hard is this to implement?
